smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,687 @@
+        logger,
+    )
+
+
+class ManifestEventHandler(PatternMatchingEventHandler):
+    """The ManifestEventHandler monitors an experiment for changes and updates
+    a telemetry datastore as needed.
+
+    It contains event handlers that are triggered by changes to a runtime experiment
+    manifest. The runtime manifest differs from a standard manifest. A runtime manifest
+    may contain multiple experiment executions in a `runs` collection.
+
+    It also contains a long-polling loop that checks experiment entities for updates
+    at each timestep."""
+
+    def __init__(
+        self,
+        pattern: str,
+        logger: logging.Logger,
+        ignore_patterns: t.Any = None,
+        ignore_directories: bool = True,
+        case_sensitive: bool = False,
+    ) -> None:
+        super().__init__(
+            [pattern], ignore_patterns, ignore_directories, case_sensitive
+        )  # type: ignore
+        self._logger = logger
+        self._tracked_runs: t.Dict[int, Run] = {}
+        self._tracked_jobs: t.Dict[_JobKey, JobEntity] = {}
+        self._completed_jobs: t.Dict[_JobKey, JobEntity] = {}
+        self._launcher_type: str = ""
+        self._launcher: t.Optional[Launcher] = None
+        self.job_manager: JobManager = JobManager(threading.RLock())
+        self._launcher_map: t.Dict[str, t.Type[Launcher]] = {
+            "slurm": SlurmLauncher,
+            "pbs": PBSLauncher,
+            "cobalt": CobaltLauncher,
+            "lsf": LSFLauncher,
+            "local": LocalLauncher,
+        }
+
+    def init_launcher(self, launcher: str) -> Launcher:
+        """Initialize the controller with a specific type of launcher.
+        SmartSim currently supports slurm, pbs(pro), cobalt, lsf,
+        and local launching
+
+        :param launcher: which launcher to initialize
+        :type launcher: str
+        :raises SSUnsupportedError: if a string is passed that is not
+                                    a supported launcher
+        :raises TypeError: if no launcher argument is provided.
+        """
+        if not launcher:
+            raise TypeError("Must provide a 'launcher' argument")
+
+        if launcher_type := self._launcher_map.get(launcher.lower(), None):
+            return launcher_type()
+
+        raise ValueError("Launcher type not supported: " + launcher)
+
+
+    def set_launcher(self, launcher_type: str) -> None:
+        """Set the launcher for the experiment"""
+        if launcher_type != self._launcher_type:
+            self._launcher_type = launcher_type
+            self._launcher = self.init_launcher(launcher_type)
+            self.job_manager.set_launcher(self._launcher)
+            self.job_manager.add_job_onstep_callback(track_timestep)
+
+            self.job_manager.start()
+
+    @property
+    def launcher(self) -> Launcher:
+        """Return a launcher appropriate for the experiment"""
+        if not self._launcher:
+            self._launcher = LocalLauncher()
```

Comment:
Should this also be calling `self.set_launcher("local")` just so that the JM launcher and the `self._launcher` attr do not go "out-of-sync"?

Reply:
@MattToast probably worth writing up a ticket to expose the JM's launcher as a property so that things that share a reference to a launcher (Controller/TM/etc.) can also have their own launcher property just expose the underlying launcher used by the JM

Reply:
telmon -> `set_launcher` does pass the same launcher into the job manager. once the first note above is added, we already have a shared launcher (though in the opposite direction you mentioned)